h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2244,7 +2260,176 @@ private void fitIRLSMML(Solver s) {
         }
       } catch (NonSPDMatrixException e) {
         Log.warn(LogMsg("Got Non SPD matrix, stopped."));
-        throw e; // TODO: Decide if we want to return a bad model or no model at all.
+        warn("Regression with MLE training:", "Got Non SPD matrix, stopped.");
```

Comment:
I'm not sure about this behavior. I think it might be better to end up with some model than none, but the user has to be warned that the model didn't finish training properly. If we decide to keep it, we should probably improve the message.

Reply:
When you nave Non SPD matrix, it cannot be avoided.  It means the dataset is bad the the gram matrix cannot be inverted because it is not SPD (semi-definite positive definite).

Reply:
Sure, this is fine.